src/asciidoctor-core-api.js

```diff
@@ -71,8 +71,13 @@ Opal.Asciidoctor.AbstractBlock.$$proto.delegate = function () {
   return this.$delegate();
 };
 
-Opal.Asciidoctor.AbstractBlock.$$proto.findBy = function (selector) {
-  return this.$find_by(toHash(selector));
+Opal.Asciidoctor.AbstractBlock.$$proto.findBy = function (selector, block) {
+  if (typeof block === 'function') {
+    return Opal.block_send(this, 'find_by', block, toHash(selector));
```

Comment:
Are you sure that `block` should be the first argument and the rest of the arguments after (ie `selector`) ?